base/util/src/com/netscape/cmsutil/crypto/CryptoUtil.java

```diff
@@ -2840,6 +2883,26 @@ public static String getNameFromHashAlgorithm(AlgorithmIdentifier ai)
         throw new NoSuchAlgorithmException();
     }
 
+    /**
+    * Maps from HMACAlgorithm name to JSS Provider HMAC Alg name.
+    */
+    public static String getHMACAlgName(String name) {
```

Comment:
Since we control the JSS provider we're using, we could place this inside JSS and add aliases for these non-standard names:

From [JSSProvider.java](https://github.com/dogtagpki/jss/blob/master/org/mozilla/jss/JSSProvider.java#L227):
```java
        put("Mac.HmacSHA1",
            "org.mozilla.jss.provider.javax.crypto.JSSMacSpi$HmacSHA1");
        put("Alg.Alias.Mac.Hmac-SHA1", "HmacSHA1");
``` 

So we could very easily add:

```java
        put("Alg.Alias.Mac.SHA-1-HMAC", "HmacSHA1");
        put("Alg.Alias.Mac.SHA-256-HMAC", "HmacSHA256");
        put("Alg.Alias.Mac.SHA-384-HMAC", "HmacSHA384");
        put("Alg.Alias.Mac.SHA-512-HMAC", "HmacSHA512");
```

This would lock us into using JSS without some work to switch, but we're already locked in and I don't think its a bad thing (since we control JSS)...

My 2c?

Reply:
I think I can explain my thinking on this. The original code took as one of it's constructor arguments the MessageDigest for the mac such as SHA-256. Since this code is simply using the various raw JSS HMAC algs, I needed to choose the correct one based on the originally intended MessageDigest algs. Such here that routine is simply mapping the original routine name into the proper raw HMAC alg name.  Perhaps you figured this out already. I noticed there were other mapping type methods in CryptoUtil and felt it was a decent thing to do. 

I do think your suggestion would work by creating aliases such that the mapping would not be needed, but that would require changing jss too this round :) I would favor this round to just get it done as easily as possible, with the least moving parts... but I'm open ...

Reply:
Yeah, this is fine then. I'd generally prefer we stick this in JSS long term, but I'll open that PR. :)